### PR TITLE
feat: introduced LearningHeader for frontend-app-learning

### PR DESCRIPTION
**Pull request by asadiqbal08**
fixes: https://github.com/mitodl/mitxonline/issues/252

#### Earlier Review References of @davidjoy 
[Reference-1](https://github.com/mitodl/frontend-component-header-mitol/pull/1)
[Reference-2](https://github.com/openedx/build-test-release-wg/issues/81#issuecomment-928050901)

As per some suggestion by David, We need to refactor the code in EdX Learning MFE to utilize the header from frontend-component-header instead of course-header

screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-edx-introduction-block-v1-edX-DemoX-Demo-Course-type-vertical-block-vertical-0270f6de40fc-2021-10-22-17_28_58
.

### CODE REVIEW

**Comment by asadiqbal08** [comment]
@davidjoy anything else need to address here or good to go ?

**Comment by davidjoy** [comment]
Ugh, it still seems to have some package-lock.json problems.  Try running `npm ci` locally to see the error - looks like package-lock.json doesn't like some font-awesome versions.  Does running `npm i` produce any changes to package-lock.json?  If not, we may need to just re-generate package-lock.json by deleting it and running `npm i`, but it'd be preferable not to if we can avoid it. 

Otherwise this looks good to go and as soon as we get that check to pass, we can merge.

**Review by davidjoy — CHANGES_REQUESTED** [review]
It looks like there are issues with package-lock.json - see my prior comment above.

**Comment by asadiqbal08** [comment]
Yes, I forgot to push updated `package.json`, Now I am not getting any error locally. let see over the CI pipeline here.

**Comment by asadiqbal08** [comment]
@davidjoy I updated the PR to fix issues that causing build failure in the CI pipeline. I make sure by re running the Github action on local devstack and also fixed the corresponding test failure issues. 🤞🤞

File: src/Header.jsx [hunk 1]
```diff
@@ -57,18 +63,18 @@ function Header({ intl }) {
...
     {
       type: 'item',
       href: config.LOGOUT_URL,
       content: intl.formatMessage(messages['header.user.menu.logout']),
     },
   ];
 
+  // Users should only see Order History if have a ORDER_HISTORY_URL define in the environment.
+  if (config.ORDER_HISTORY_URL) {
+    userMenu.splice(-1, 0, orderHistoryItem);
```
Comment (asadiqbal08) [inline src/Header.jsx:75]: @davidjoy FYI, I kept the OrderHistory conditional on basis of `ORDER_HISTORY_URL`

File: src/learning-header/AuthenticatedUserDropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        </Dropdown.Toggle>
+        <Dropdown.Menu className="dropdown-menu-right">
+          {dashboardMenuItem}
+          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/u/${username}`}>
+            {intl.formatMessage(messages.profile)}
+          </Dropdown.Item>
+          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/account/settings`}>
+            {intl.formatMessage(messages.account)}
+          </Dropdown.Item>
+          { getConfig().ORDER_HISTORY_URL && (
```
Comment (asadiqbal08) [inline src/learning-header/AuthenticatedUserDropdown.jsx:38]: Here too.

**Comment by edx-webhook** [comment]
@asadiqbal08 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-component-header&entry.867055334=https://github.com/edx/frontend-component-header/pull/133&entry.1484655318=https://github.com/asadiqbal08&entry.752974735=2021-10-28+08:28&entry.1917517419=2021-12-02+19:17&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**